### ISSUE TITLE
error[E0428]: the name `imp` is defined multiple times

### ISSUE THREAD

**Issue by aramd** [BurntSushi/rust-memchr#69]
> [192.168.18.146] out: error[E0428]: the name `imp` is defined multiple times
[192.168.18.146] out:    --> /home/aram/.cargo/registry/src/github.com-1ecc6299db9ec823/memchr-2.3.1/src/lib.rs:148:5
[192.168.18.146] out:     |
[192.168.18.146] out: 139 |     fn imp(n1: u8, haystack: &[u8]) -> Option<usize> {
[192.168.18.146] out:     |     ------------------------------------------------ previous definition of the value `imp` here
[192.168.18.146] out: ...
[192.168.18.146] out: 148 |     fn imp(n1: u8, haystack: &[u8]) -> Option<usize> {
[192.168.18.146] out:     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `imp` redefined here
[192.168.18.146] out:     |
[192.168.18.146] out:     = note: `imp` must be defined only once in the value namespace of this block

Seems **[cfg(all(target_arch = "x86_64", memchr_runtime_simd, not(miri)))]** and **[cfg(all(memchr_libc, not(all(target_arch = "x86_64", memchr_runtime_simd, miri))))]** are not mutually exclusive.

**Comment by Ten0** [comment]
Duplicate of #68. (A few seconds aside ^^)

**Comment by BurntSushi** [comment]
I'ved yanked `2.3.1` from crates.io until I have time to properly investigate this. cc @ldesgoui

**Comment by BurntSushi** [comment]
(It was almost certainly caused by #67.)

**Comment by BurntSushi** [comment]
Closing as duplicate of #68.
